examples/codon.ipynb

```diff
@@ -0,0 +1,213 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SynBioPython codon usage module"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook demonstrates the functionalities of the codon usage module of [SynBiopython](https://synbiopython.org)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from collections import Counter\n",
+    "\n",
+    "from synbiopython import codon\n",
+    "from synbiopython.codon import table, utils"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The codon module supports both organism names and taxonomy ids, and mapping between both identifiers:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Name: Saccharomyces cerevisiae\n",
+      "Taxonomy id: 4932\n"
+     ]
+    }
+   ],
+   "source": [
+    "name = codon.get_name(4932)\n",
+    "tax_id = codon.get_tax_id(\"Saccharomyces cerevisiae\")\n",
+    "\n",
+    "print(\"Name:\", name)\n",
+    "print(\"Taxonomy id:\", tax_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Codon usage tables can be retrieved through either name or taxonomy id:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "name_table = table.get_table(name)\n",
+    "tax_id_table = table.get_table(tax_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "These will be the same, irrespective of the method of retrieval:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "assert name_table == tax_id_table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The table is a simple dictionary of amino acids to codons, which are themselves a codon to usage frequency dictionary:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'CTC': 0.06, 'CTG': 0.11, 'CTT': 0.13, 'CTA': 0.14, 'TTA': 0.28, 'TTG': 0.29}\n"
+     ]
+    }
+   ],
+   "source": [
+    "l_codons = name_table[\"L\"]\n",
+    "print(l_codons)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Utility methods"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Utility methods are available for randomly sample codons, and for performing simple codon optimisations.\n",
```

Comment:
"randomly sampling" ?

Reply:
Whoops. Thanks. And I'm supposed to be a native speaker!